doc/devel/development_workflow.rst

```diff
@@ -452,48 +452,56 @@ Automated tests
+  locally.
 
+  .. _tox: https://tox.readthedocs.io/
 
 * Codecov and CodeQL are currently for information only. Their failure is not
   necessarily a blocker.
 
-* tox_ is not used in the automated testing. It is supported for testing
-  locally.
+Make sure the Linting, GitHub Actions, AppVeyor, CircleCI, and Azure pipelines are
+passing before merging. All checks are listed at the bottom of the GitHub page of your
+pull request.
 
-  .. _tox: https://tox.readthedocs.io/
+Here are some tips for finding the cause of a test failure:
+
+- If *Linting* fails, you have a code style issue, which will be listed
+  as annotations on the pull request's diff.
+- If *Mypy* or *Stubtest* fails, you have inconsistency in type hints, which
+  will be listed as annotations in the diff.
+- If a GitHub Actions or AppVeyor run fails, search the log for ``FAILURES``.
+  The subsequent section will contain information on the failed tests.
+- If CircleCI fails, likely you have some reStructuredText style issue in
+  the docs. Search the CircleCI log for ``WARNING``.
+- If Azure pipelines fail with an image comparison error, you can find the
+  images as *artifacts* of the Azure job:
+
+  - Click *Details* on the check on the GitHub PR page.
+  - Click *View more details on Azure Pipelines* to go to Azure.
+  - On the overview page *artifacts* are listed in the section *Related*.
+
+Skip CI checks
+--------------
+
+If you know only a subset of CIs need to be run, this can be controlled on individual
+commits by including the following substrings in commit messages:
+
+- ``[ci doc]``: restrict the CI to documentation checks. For when you only changed
+  documentation. This skip is automatic if the changes are only under
+  ``doc/`` or ``galleries/``.
+- ``[skip circle]``: skip the documentation build check. For when you didn't
+  change documentation.
+- Unit tests can be turned off for individual platforms with:
 
-* If you know only a subset of CIs need to be run, this can be controlled on
-  individual commits by including the following substrings in commit messages:
+  - ``[skip actions]``: GitHub Actions
+  - ``[skip appveyor]``: AppVeyor. Must be in the first line of the commit message.
+  - ``[skip azp]``: Azure Pipelines
 
-  - ``[ci doc]``: restrict the CI to documentation checks. For when you only
-    changed documentation (this skip is automatic if the changes are only under
-    ``doc/`` or ``galleries/``).
-  - ``[skip circle]``: skip the documentation build check. For when you didn't
-    change documentation.
-  - Unit tests can be turned off for individual platforms with
+- ``[skip ci]``: skip all CIs.  Use this only if you know your changes do not
+  need to be tested at all, which is very rare.
 
-    - ``[skip actions]``: GitHub Actions
-    - ``[skip appveyor]`` (must be in the first line of the commit): AppVeyor
-    - ``[skip azp]``: Azure Pipelines
+Skip instructions only apply to ``on: push`` and ``on: pull_request`` events. For more
```

Comment:
I want to be a little careful here... these terms and the link provided only apply to github actions, not appveyor/azp/circle...